### PR TITLE
Retrieve MIME type viewer config

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

This adds a new endpoint (`/api/mimetype/viewerconfig/{type}/{sub}`) to support the retrieval of the configuration of viewers for a MIME type. This information can then be used in the front-end to support viewing attachments from the New Search UI - and later, item summary pages and the like.

To include tests, I've moved MimeTypeApiTest into the new io.github.openequella setup. This allows the existing test to run much faster, plus makes it easier to add the additional test.

(Also snuck in a tweak for editor config so that it matches our auto formatters.)

image

#1306

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+public class MimeTypeApiTest extends AbstractRestApiTest {
+
+  private static final String MIMETYPE_API_ENDPOINT =
+      TEST_CONFIG.getInstitutionUrl() + "api/mimetype";
+
+  @Test
+  public void testRetrieveMimeTypes() throws Exception {
+    final List<MimeTypeDetail> initialFilters = getMimeTypes();
+    assertEquals(153, initialFilters.size());
```
Comment (mrblippy) [inline autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:23]: So, the number 153 is the total number of mimetypes in the institution. 
Could that number be placed it a named variable to aid in readability?

**Review by mrblippy — APPROVED** [review]
Just one tiny, tiny readbility nitpick. But apart from that it looks good!

File: autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java (hunk 1 above)
Comment (edalex-ian) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:23]: Hmmm, this was the original test which I moved across. Ummm, well a variable could be OTT, but at the very least a magic number should be documented in some form - you are correct.

assertEquals (like many others) allows you to specify a message. So I'll change this to:

```java
    assertEquals("The number of returned filters should match the institution total.", 153,
      initialFilters.size());
```

Which then provides a form of documentation, and also on failure would nicely report like:

```
java.lang.AssertionError: The number of returned filters should match the institution total. 
Expected :152
Actual   :153
```

Comment (mrblippy) [reply autotest/OldTests/src/test/java/io/github/openequella/rest/MimeTypeApiTest.java:23]: Yeah, i think that's a good approach :+1: :+1:

**Review by SammyIsConfused — APPROVED** [review]
Looks good to me!

**Comment by edalex-ian** [comment]
I'll keep my fingers crossed and wait for a green build to merge. :crossed_fingers:

**Pull request merged**